### PR TITLE
Add AirLoopHVAC to ThreeJS user data

### PR DESCRIPTION
**Pull request by antoine-galataud**
Pull request overview
---------------------

First step in implementation of https://github.com/openstudiocoalition/OpenStudioApplication/issues/413

### Pull Request Author

 - [ ] Model API Changes / Additions
 - [ ] Any new or modified fields have been implemented in the EnergyPlus ForwardTranslator (and ReverseTranslator as appropriate)
 - [ ] Model API methods are tested (in `src/model/test`)
 - [ ] EnergyPlus ForwardTranslator Tests (in `src/energyplus/Test`)
 - [ ] If a new object or method, added a test in NREL/OpenStudio-resources: Add Link
 - [ ] If needed, added VersionTranslation rules for the objects (`src/osversion/VersionTranslator.cpp`)
 - [ ] Verified that C# bindings built fine on Windows, partial classes used as needed, etc.
 - [X] All new and existing tests passes
 - [ ] If methods have been deprecated, update rest of code to use the new methods

**Labels:**

 - [ ] If change to an IDD file, add the label `IDDChange`
 - [ ] If breaking existing API, add the label `APIChange`
 - [ ] If deemed ready, add label `Pull Request - Ready for CI` so that CI builds your PR

### Review Checklist

This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] Code Style, strip trailing whitespace, etc.
 - [ ] All related changes have been implemented: model changes, model tests, FT changes, FT tests, VersionTranslation, OS App
 - [ ] Labeling is ok
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified

### CODE REVIEW

**Comment by antoine-galataud** [comment]
Builds failed but I can't access the CI tool (times out on every attempt). Is there an other convenient way to obtain build results?

**Comment by tijcolem** [comment]
@antoine-galataud I just emailed you concerning build logs.  

This is the snippet of the error message for osx-incremental

```
/Users/jenkins/git/OpenStudioIncremental/develop/src/utilities/geometry/Test/ThreeJS_GTest.cpp:56:30: error: loop variable 'sceneChild' of type 'const openstudio::ThreeSceneChild' creates a copy from type 'const openstudio::ThreeSceneChild' [-Werror,-Wrange-loop-analysis]

  for (const ThreeSceneChild sceneChild : sceneChildren) {

                             ^

/Users/jenkins/git/OpenStudioIncremental/develop/src/utilities/geometry/Test/ThreeJS_GTest.cpp:56:8: note: use reference type 'const openstudio::ThreeSceneChild &' to prevent copying

  for (const ThreeSceneChild sceneChild : sceneChildren) {

       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

                             &

/Users/jenkins/git/OpenStudioIncremental/develop/src/utilities/geometry/Test/ThreeJS_GTest.cpp:62:39: error: loop variable 'metadata' of type 'const openstudio::ThreeModelObjectMetadata' creates a copy from type 'const openstudio::ThreeModelObjectMetadata' [-Werror,-Wrange-loop-analysis]

  for (const ThreeModelObjectMetadata metadata : scene->metadata().modelObjectMetadata()) {

                                      ^

/Users/jenkins/git/OpenStudioIncremental/develop/src/utilities/geometry/Test/ThreeJS_GTest.cpp:62:8: note: use reference type 'const openstudio::ThreeModelObjectMetadata &' to prevent copying

  for (const ThreeModelObjectMetadata metadata : scene->metadata().modelObjectMetadata()) {

       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

                                      &

2 errors generated.

make[2]: *** [src/utilities/CMakeFiles/openstudio_utilities_tests.dir/geometry/Test/ThreeJS_GTest.cpp.o] Error 1

make[2]: *** Waiting for unfinished jobs....

[ 23%] Building CXX object src/model/CMakeFiles/openstudio_model_tests.dir/test/ThreeJSReverseTranslator_GTest.cpp.o

make[1]: *** [src/utilities/CMakeFiles/openstudio_utilities_tests.dir/all] Error 2

make[1]: *** Waiting for unfinished jobs....

Consolidate compiler generated dependencies of target openstudio_radiance_tests

```

**Comment by jmarrec** [comment]
@tijcolem I already added review suggest for these unecessary copies, but I forgot to post my review.. so it was only visible on /files

**Comment by jmarrec** [comment]
osx has a failing test but that's irrelevant here

```
[ RUN      ] BCLFixture.BCLComponent

/Users/jenkins/git/OpenStudioIncremental/develop/src/utilities/bcl/test/BCLComponent_GTest.cpp:44: Failure

Expected equality of these values:

  "philadelphia pa [724086 TMY2-13739]"

  component.name()

    Which is: ""

```

**Comment by tijcolem** [comment]
cool, yep that failing test is unrelated to this PR.  Need to check on windows but Linux and Mac seems happy now.

**Comment by antoine-galataud** [comment]
Thanks @tijcolem and @jmarrec for reviewing and fixing.

File: src/utilities/geometry/Test/ThreeJS_GTest.cpp [hunk 1]
```diff
@@ -51,6 +51,19 @@ TEST_F(GeometryFixture, ThreeJS) {
...
+  for (const ThreeSceneChild sceneChild : sceneChildren) {
+    EXPECT_DOUBLE_EQ(1, sceneChild.userData().airLoopHVACNames().size());
+  }
+
+  // checking metadata
+  bool alhvac_metadata_found = false;
+  for (const ThreeModelObjectMetadata metadata : scene->metadata().modelObjectMetadata()) {
+    alhvac_metadata_found |= istringEqual("OS:AirLoopHVAC", metadata.iddObjectType()) && istringEqual("Air Loop HVAC 1", metadata.name());
+  }
+  ASSERT_TRUE(alhvac_metadata_found);
```
[the remark this answers is not in the document]
Comment (antoine-galataud) [reply src/utilities/geometry/Test/ThreeJS_GTest.cpp]: nice construct, although makes it a bit harder to read IMHO. But my C++ skills are a bit rusted ;)

File: src/utilities/geometry/Test/ThreeJS_GTest.cpp [hunk 2]
```diff
@@ -51,6 +51,19 @@ TEST_F(GeometryFixture, ThreeJS) {
   boost::optional<ThreeScene> scene = ThreeScene::load(contents);
   ASSERT_TRUE(scene);
 
+  // checking user data
+  std::vector<ThreeSceneChild> sceneChildren = scene->object().children();
+  for (const ThreeSceneChild& sceneChild : sceneChildren) {
+    EXPECT_DOUBLE_EQ(1, sceneChild.userData().airLoopHVACNames().size());
```
Comment (tijcolem) [inline src/utilities/geometry/Test/ThreeJS_GTest.cpp:57]: Can we change to `EXPECT_DOUBLE_EQ(1, double(sceneChild.userData().airLoopHVACNames().size()));`

windows is throwing the following due to treating warning as error when compiling. 

```
D:\jenkins\openstudio\develop\src\utilities\geometry\Test\ThreeJS_GTest.cpp(57,5): error C2220: the following warning is treated as an error [D:\jenkins\openstudio\develop\build\src\utilities\openstudio_utilities_tests.vcxproj]
D:\jenkins\openstudio\develop\src\utilities\geometry\Test\ThreeJS_GTest.cpp(57,5): warning C4244: 'argument': conversion from 'unsigned __int64' to 'RawType', possible loss of data [D:\jenkins\openstudio\develop\build\src\utilities\openstudio_utilities_tests.vcxproj]
```

Comment (jmarrec) [reply src/utilities/geometry/Test/ThreeJS_GTest.cpp]: Nah, you want `EXPECT_EQ(1, xxx.size());` (or `EXPECT_EQ(1u, xxx.size());` )

File: src/utilities/geometry/Test/ThreeJS_GTest.cpp [hunk 3]
```diff
@@ -51,6 +51,19 @@ TEST_F(GeometryFixture, ThreeJS) {
   boost::optional<ThreeScene> scene = ThreeScene::load(contents);
   ASSERT_TRUE(scene);
 
+  // checking user data
+  std::vector<ThreeSceneChild> sceneChildren = scene->object().children();
+  for (const ThreeSceneChild& sceneChild : sceneChildren) {
+    EXPECT_DOUBLE_EQ(1, double(sceneChild.userData().airLoopHVACNames().size()));
```
Comment (jmarrec) [inline src/utilities/geometry/Test/ThreeJS_GTest.cpp:57]:
```suggestion
    EXPECT_EQ(1, sceneChild.userData().airLoopHVACNames().size());
```

**Review by jmarrec — CHANGES_REQUESTED** [review]

**Review by jmarrec — APPROVED** [review]

File: src/utilities/geometry/Test/ThreeJS_GTest.cpp (hunk 2 above)
Comment (tijcolem) [reply src/utilities/geometry/Test/ThreeJS_GTest.cpp]: Even better!

**Review by tijcolem — APPROVED** [review]
Thanks @antoine-galataud for making those changes. I looked CI and Ubuntu 20.04 and osx are unrelated so this looks good to drop in!

**Pull request merged**